ypkg2/compressdoc.py

```diff
@@ -0,0 +1,120 @@
+#!/bin/true
+# -*- coding: utf-8 -*-
+#
+#  This file is part of ypkg2
+#
+#  Copyright 2022 Solus Project
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+
+import gzip
+import os
+
+compressed_exts = [
+    ".gz",
+    ".bz2",
+    ".zst"
```

Comment:
.xz and lzip (.lz?) are also supported https://gitlab.com/cjwatson/man-db/-/blob/main/configure.ac#L394